splunk/debian-9/Dockerfile

```diff
@@ -11,61 +11,98 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM base-debian-9:latest
 
+
+#
+# Download and unpack Splunk Enterprise
+#
+FROM base-debian-9:latest as package
 ARG SPLUNK_FILENAME
 ARG SPLUNK_BUILD_URL
-ARG SPLUNK_DEFAULTS_URL
+COPY splunk/debian-9/splunk-minimal-exclude.list /tmp
+RUN echo "Downloading Splunk and validating the checksum at: ${SPLUNK_BUILD_URL}"
+RUN wget -qO /tmp/${SPLUNK_FILENAME} ${SPLUNK_BUILD_URL}
+RUN wget -qO /tmp/${SPLUNK_FILENAME}.sha512 ${SPLUNK_BUILD_URL}.sha512
+RUN (cd /tmp && sha512sum -c ${SPLUNK_FILENAME}.sha512)
+RUN mkdir -p /opt/splunk/var /minimal/splunk/var /extras/splunk/var
+RUN tar -C /minimal/splunk --strip 1 --exclude-from=/tmp/splunk-minimal-exclude.list -zxf /tmp/${SPLUNK_FILENAME}
+RUN tar -C /extras/splunk --strip 1 --wildcards --files-from=/tmp/splunk-minimal-exclude.list -zxf /tmp/${SPLUNK_FILENAME}
+RUN mv /minimal/splunk/etc /minimal/splunk-etc
+RUN mv /extras/splunk/etc /extras/splunk-etc
+RUN mkdir -p /minimal/splunk/etc/apps/_splunk_config /extras/splunk/etc/apps/_splunk_config
+
 
-ENV SPLUNK_HOME=/opt/splunk \
+#
+# Minimal Splunk base image with many files excluded, intended for internal and experimental use
+#
+FROM base-debian-9:latest as minimal-debian-9
+LABEL maintainer="support@splunk.com"
+ENV TZ=UTC \
+    SPLUNK_HOME=/opt/splunk \
     SPLUNK_GROUP=splunk \
     SPLUNK_USER=splunk \
-    SPLUNK_ROLE=splunk_standalone \
-    SPLUNK_FILENAME=${SPLUNK_FILENAME} \
+    TMPSPLUNKDIR=${SPLUNK_HOME}/tmp \
+    TMPETCDIR=${TMPSPLUNKDIR}/etc
+
+# Currently kubernetes only accepts UID and not USER field to
+# start a container as a particular user. So we create Splunk
+# user with pre-determined UID.
+ARG UID=41812
+ARG GID=41812
+    
+RUN ln -sf /usr/share/zoneinfo/${TZ} /etc/localtime \
```

Comment:
Is this a specific requirement for internal use? Or would this be more apt to build into our `base-x-x` images?

Reply:
Similar to UID/GID, I'm kind of split on these changes. Could be good to standardize on these things across the board, but also easy to change them in derived containers, if desired..